filebeat/crawler/prospector.go

```diff
@@ -71,21 +74,28 @@ func (p *Prospector) Init() error {
 }
 
 // Starts scanning through all the file paths and fetch the related files. Start a harvester for each file
-func (p *Prospector) Run() {
+func (p *Prospector) Run(wg *sync.WaitGroup) {
+
+	// TODO: Defer the call to block shutdown
+	wg.Done()
```

Comment:
@urso Would you agree with this temporary hack? So shutdown of prospector is non blocking. This would be resolved as soon as the proper harvester shutdown would be implemented. The current behaviour would be almost identical to before but has all the structure ready to implement it properly.